myia/operations/op_full.py

```diff
@@ -0,0 +1,45 @@
+"""Implementation of operation for numpy.full."""
+
+import numpy as np
+
+from ..lib import (
+    abstract_array,
+    core,
+    distribute,
+    myia_to_array,
+    scalar_cast,
+    to_scalar_type,
+    typeof,
+)
+
+
+def pyimpl_full(shape, fill_value, dtype):
+    """Python implementation for operation full."""
+    return np.full(shape, fill_value, dtype)
+
+
+@core
+def full(shape, fill_value, dtype):
+    """
+    Main code for operation full.
+
+    :param shape: a tuple of integers
+    :param fill_value: a scalar value
+    :param dtype: either a string (e.g. 'int32')
+        or a numpy dtype (e.g. np.int32)
+    :return: an array
+    """
+    abstract_scalar_type = to_scalar_type(dtype, typeof(fill_value))
```

Comment:
I'm not a huge fan of the `default` argument on `to_scalar_type`. I think it would be better to do it like this:

```suggestion
    if dtype is None:
        dtype = typeof(fill_value)
    abstract_scalar_type = to_scalar_type(dtype)
```

Reply:
Done!